File: update-galaxy-location.py
import firebase_admin
from firebase_admin import credentials, firestore
import os

# Initialize Firebase Admin SDK
cred = credentials.ApplicationDefault()
firebase_admin.initialize_app(cred, {
    'projectId': 'plane-alert-800ff',
})

db = firestore.client()

device_id = 'u4h7b5hnvdgvozqd5yzm86i474fs4g__galaxys24'

# Dieselstraße 8, Unterföhring, 85774 coordinates  
new_location = {
    'lat': 48.1896,
    'lon': 11.6490
}

try:
    # Get current device data
    doc_ref = db.collection('devices').document(device_id)
    doc = doc_ref.get()
    
    if not doc.exists:
        print(f'❌ Device not found: {device_id}')
    else:
        current_data = doc.to_dict()
        print('📱 Current device data:')
        print(f'   Device: {current_data.get("deviceName")}')
        print(f'   Current location: {current_data.get("location")}')
        
        # Update location
        doc_ref.update({
            'location': new_location,
            'updatedAt': firestore.SERVER_TIMESTAMP
        })
        
        print('\n✅ Location updated successfully!')
        print(f'   New location: {new_location}')
        print('   Address: Dieselstraße 8, Unterföhring, 85774')
        
except Exception as error:
    print(f'❌ Error updating location: {error}')

File: update-galaxy-location.js
// Update Galaxy S24 location to Unterföhring
const admin = require('firebase-admin');
const path = require('path');

// Initialize Firebase Admin
const serviceAccount = require(path.join(__dirname, 'plane-alert-800ff-firebase-adminsdk-8tgav-00c8aae60f.json'));

admin.initializeApp({
  credential: admin.credential.cert(serviceAccount),
  projectId: 'plane-alert-800ff'
});

const db = admin.firestore();

async function updateGalaxyLocation() {
  const deviceId = 'u4h7b5hnvdgvozqd5yzm86i474fs4g__galaxys24';
  
  // Dieselstraße 8, Unterföhring, 85774 coordinates
  const newLocation = {
    lat: 48.1896,
    lon: 11.6490
  };

  try {
    // Get current device data
    const docRef = db.collection('devices').doc(deviceId);
    const doc = await docRef.get();
    
    if (!doc.exists) {
      console.log('❌ Device not found:', deviceId);
      return;
    }

    console.log('📱 Current device data:');
    const currentData = doc.data();
    console.log('   Device:', currentData.deviceName);
    console.log('   Current location:', currentData.location);
    
    // Update location
    await docRef.update({
      location: newLocation,
      updatedAt: admin.firestore.FieldValue.serverTimestamp()
    });

    console.log('\n✅ Location updated successfully!');
    console.log('   New location:', newLocation);
    console.log('   Address: Dieselstraße 8, Unterföhring, 85774');
    
  } catch (error) {
    console.error('❌ Error updating location:', error);
  } finally {
    process.exit(0);
  }
}

updateGalaxyLocation();
